### ISSUE TITLE
[@types/gradient-parser] TS2339: Property 'length' does not exist on type 'HexNode'.

### ISSUE THREAD

**Issue by deadcoder0904** [DefinitelyTyped/DefinitelyTyped#52612]
I tried using the `@types/gradient-parser` package and had a simple problem.

I have the following code:

```tsx
import gradient from 'gradient-parser'

const parsedLinearGradient = gradient.parse(selectedColor.linearGradient.color)

parsedLinearGradient[0].colorStops.forEach((c) => {
	if (c.type === 'hex') {
		defaultColors.push(`#${c.value}`)
		defaultPositions.push(c.length.value)
	}
})
```

And it gets red-squiggly lines on `length`. It says: `TS2339: Property 'length' does not exist on type 'HexNode'.`

Would love a fix @AviVahl :)

**Comment by AviVahl** [comment]
Feel free to make a PR. The package types were reversed engineered from its README and tests.
It may be that some specific cases aren't covered correctly, if they weren't covered in the above.

If you see `.length.value` available in runtime, adjust the types and tests to reflect that possible structure.

**Comment by deadcoder0904** [comment]
According to,

https://github.com/rafaelcaricio/gradient-parser/blob/cce801d1df27261f998230bec29f0d8c8eeb682a/spec/parser.spec.js#L108-L132

and

https://github.com/rafaelcaricio/gradient-parser/blob/cce801d1df27261f998230bec29f0d8c8eeb682a/lib/parser.js#L259-L264

I think it needs an optional `length` type on `HexNode`, `RgbNode` and `RgbaNode`.

There might be more TS updates as the library recently got updated but this should suffice for now.

Fix #52623 is ready to merge 🚀
